init.lua

```diff
@@ -910,57 +908,61 @@ if progressive_mode then
 			save_meta(players[i])
 		end
 	end)
-else
-	mt.register_chatcommand("craft", {
-		description = S("Show recipe(s) of the pointed node"),
-		func = function(name)
-			local player = mt.get_player_by_name(name)
-			local ppos   = player:get_pos()
-			local dir    = player:get_look_dir()
-			local eye_h  = {x = ppos.x, y = ppos.y + 1.625, z = ppos.z}
-			local node_name
-
-			for i = 1, 10 do
-				local look_at = vector_add(eye_h, vector_mul(dir, i))
-				local node = mt.get_node(look_at)
-
-				if node.name ~= "air" then
-					node_name = node.name
-					break
-				end
-			end
+end
 
-			if not node_name then
-				return false, mt.colorize("red", "[craftguide] ") ..
-						S("No node pointed")
+mt.register_chatcommand("craft", {
+	description = S("Show recipe(s) of the pointed node"),
+	func = function(name)
+		local player = mt.get_player_by_name(name)
+		local ppos   = player:get_pos()
+		local dir    = player:get_look_dir()
+		local eye_h  = {x = ppos.x, y = ppos.y + 1.625, z = ppos.z}
+		local node_name
+
+		for i = 1, 10 do
+			local look_at = vector_add(eye_h, vector_mul(dir, i))
+			local node = mt.get_node(look_at)
+
+			if node.name ~= "air" then
+				node_name = node.name
+				break
 			end
+		end
 
-			local data = player_data[name]
-			reset_data(data)
-
-			local recipes = recipes_cache[node_name]
-			local is_fuel = fuel_cache[node_name]
-
-			if not recipes then
-				if is_fuel then
-					recipes = get_item_usages(node_name)
-					if #recipes > 0 then
-						data.show_usages = true
-					end
-				else
-					return false, mt.colorize("red", "[craftguide] ") ..
-						S("No recipe for this node:") .. " " ..
-						mt.colorize("yellow", node_name)
+		if not node_name then
+			return false, mt.colorize("red", "[craftguide] ") ..
+					S("No node pointed")
+		end
+
+		local data = player_data[name]
+		reset_data(data)
+
+		local recipes = recipes_cache[node_name]
+		local is_fuel = fuel_cache[node_name]
+
+		if recipes then
+			recipes = apply_recipe_filters(recipes, player)
+		end
+
+		if not recipes or #recipes == 0 then
```

Comment:
Btw we need some standardization to check an empty array. Either we pick `next(table)` or `#recipes == 0` but no both mixed in the codebase.

Reply:
`#recipes` is faster, but it can only be used for array tables.